pixeltable/functions/huggingface.py

```diff
@@ -282,6 +282,80 @@ def detr_for_object_detection(image: Batch[PIL.Image.Image], *, model_id: str, t
     ]
 
 
+@pxt.udf(batch_size=4)
+def vit_for_image_classification(
+    image: Batch[PIL.Image.Image],
+    *,
+    model_id: str,
+    top_k: int = 5
+) -> Batch[list[dict[str, Any]]]:
+    """
+    Computes image classifications for the specified image using a Vision Transformer (ViT) model.
+    `model_id` should be a reference to a pretrained [ViT Model](https://huggingface.co/docs/transformers/en/model_doc/vit).
+
+    __Note:__ Be sure the model is a ViT model that is trained for image classification, such as
+    `google/vit-base-patch16-224`. General feature-extraction models such as `google/vit-base-patch16-224-in21k`
+    will not produce the desired results.
+
+    __Requirements:__
+
+    - `pip install transformers`
+
+    Args:
+        image: The image to classify.
+        model_id: The pretrained model to use for the classification.
+        top_k: The number of classes to return.
+
+    Returns:
+        A list of the `top_k` highest-scoring classes for each image. Each element in the list is a dictionary
+            in the following format:
+
+    ```python
+    {
+        'p': 0.494,  # class probability
+        'class': 935,  # class ID
+        'label': 'mashed potato',  # class label
+    }
+    ```
+
+    Examples:
+        Add a computed column that applies the model `google/vit-base-patch16-224` to an existing
+        Pixeltable column `tbl.image` of the table `tbl`:
+
+        >>> tbl['image_class'] = vit_for_image_classification(
+        ...     tbl.image, model_id='google/vit-base-patch16-224'
+        ... )
+    """
+    env.Env.get().require_package('transformers')
+    device = resolve_torch_device('auto')
+    import torch
+    from transformers import ViTImageProcessor, ViTForImageClassification
+
+    model: ViTForImageClassification = _lookup_model(model_id, ViTForImageClassification.from_pretrained, device=device)
+    processor = _lookup_processor(model_id, ViTImageProcessor.from_pretrained)
+    normalized_images = [normalize_image_mode(img) for img in image]
```

Comment:
We need to get away from using this pattern (cpu-intensive operations that create shareable intermediate state) and instead switch to expr_udfs. Happy to discuss.